packages/gatsby-theme-project-portal/src/layouts/SearchPageLayout.tsx

```diff
@@ -0,0 +1,265 @@
+        faq: {
+          text: string
+          title: string
+        }[]
+        aims: {
+          text: string
+          title: string
+        }[]
+        title: string
+      }
+    }
+  }
+}
+
+export const SearchPageLayout: FunctionComponent<SearchProps> = ({
+  data: {
+    site: {
+      siteMetadata: { siteUrl },
+    },
+    allProject,
+    allGeneralPage,
+  },
+}: SearchProps) => {
+  const [searchQuery, setSearchQuery] = useState([])
+  const [queryResults, setQueryResults] = useState([])
+  const searchNodes = { siteUrl, allProject, allGeneralPage }
+  let documents = []
+
+  Object.keys(searchNodes).forEach((page) => {
+    if (page === "siteUrl") {
+      return
+    } else {
+      searchNodes[page].nodes.forEach((node) => {
+        let tempNode = structuredClone(node)
+        const newItem = Object.values(tempNode).map((field) => {
+          if (isNull(field)) {
+            return null
+          }
+
+          if (typeof field === "object") {
+            return Object.values(field)
+              .filter((value) => !isNull(value))
+              .map((value) =>
+                typeof value === "string"
+                  ? value
+                  : Object.values(value)
+                      .filter((k) => !isNull(k))
+                      .join(" ")
+              )
+              .join(" ")
+          } else {
+            return field
+          }
+        })
+        if (page === "allProject") {
+          if (!node.slug.includes("/")) {
+            tempNode.slug = `project/${node.slug}`
+          }
+        }
+
+        Object.keys(tempNode).forEach((page, i) => {
+          tempNode[page] = newItem[i]
+        })
+        documents.push(tempNode)
+      })
+    }
+  })
+
+  let db = documents.reduce(function (acc, document) {
+    acc[document.slug] = document
+    return acc
+  }, {})
+
+  let idx = lunr(function () {
+    this.ref("slug") // identifier
+    // fields to search
+    this.field("slug")
+    this.field("faq")
+    this.field("aims")
+    this.field("lede")
+    this.field("title")
+    this.field("question")
+    this.field("status")
+    this.field("agency")
+    this.field("topics")
+    this.field("mainContact")
+    this.field("summary")
+    this.field("status")
+    this.field("purpose")
+    this.field("emailContent")
+    this.field("projectTeam")
+    this.field("expertise")
+    this.field("requirement")
+
+    // allow access to the position of the found word in the field
+    this.metadataWhitelist = ["position"]
+    /* allow stop words (see https://github.com/olivernn/lunr.js/blob/master/lunr.js)
+     we have 'about' as one of the slugs, which spurred this decision
+    We can implement our own stopWordFilter:
+    var idx = lunr(function () {
+      // normal index definition
+      this.pipeline.after(lunr.stopWordFilter, myStopWordFilter)
+       this.pipeline.remove(lunr.stopWordFilter)
+    })
+     */
+    this.pipeline.remove(lunr.stopWordFilter)
```

Comment:
could we remove "about" from the lunr.stopWordFilter and still have the benefit of the filter? Doesn't look too complicated to reuse: https://lunrjs.com/docs/stop_word_filter.js.html

Reply:
@hollandjg Are there any other words within the general stop word list that we should re-add?

[
    "a",
    "able",
    "across",
    "after",
    "all",
    "almost",
    "also",
    "am",
    "among",
    "an",
    "and",
    "any",
    "are",
    "as",
    "at",
    "be",
    "because",
    "been",
    "but",
    "by",
    "can",
    "cannot",
    "could",
    "dear",
    "did",
    "do",
    "does",
    "either",
    "else",
    "ever",
    "every",
    "for",
    "from",
    "get",
    "got",
    "had",
    "has",
    "have",
    "he",
    "her",
    "hers",
    "him",
    "his",
    "how",
    "however",
    "i",
    "if",
    "in",
    "into",
    "is",
    "it",
    "its",
    "just",
    "least",
    "let",
    "like",
    "likely",
    "may",
    "me",
    "might",
    "most",
    "must",
    "my",
    "neither",
    "no",
    "nor",
    "not",
    "of",
    "off",
    "often",
    "on",
    "only",
    "or",
    "other",
    "our",
    "own",
    "rather",
    "said",
    "say",
    "says",
    "she",
    "should",
    "since",
    "so",
    "some",
    "than",
    "that",
    "the",
    "their",
    "them",
    "then",
    "there",
    "these",
    "they",
    "this",
    "tis",
    "to",
    "too",
    "twas",
    "us",
    "wants",
    "was",
    "we",
    "were",
    "what",
    "when",
    "where",
    "which",
    "while",
    "who",
    "whom",
    "why",
    "will",
    "with",
    "would",
    "yet",
    "you",
    "your",
  ])